pkg/monitortests/kubeapiserver/legacykubeapiservermonitortests/early_e2e_disruption.go

```diff
@@ -0,0 +1,65 @@
+package legacykubeapiservermonitortests
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"regexp"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+// amount of time after E2E tests start that we consider "early"
+const earlyE2ECheckDuration = 3 * time.Minute
+
+func testEarlyE2EAPIServerDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[Jira:\"kube-apiserver\"] API servers should not experience disruption near the start of E2E testing"
+
+	// regex to match the backends we care about
+	var earlyE2EBackendRe = regexp.MustCompile(`^(cache-)?(openshift|kube|oauth)-api-`)
+
+	var periodStart, periodEnd time.Time // the time period we're interested in disruption
+	for _, event := range events {
+		if event.Source == monitorapi.SourceE2ETest {
+			periodStart = event.From
+			periodEnd = event.From.Add(earlyE2ECheckDuration)
+			break
+		}
+	}
+
+	junits := []*junitapi.JUnitTestCase{}
+	if periodStart.IsZero() {
+		return junits // no e2e tests ran, no point in this test
+	}
+	intervalCount := 0
+	intervalMessages := sets.New[string]()
+	for _, event := range events {
+		if event.Source != monitorapi.SourceDisruption || event.Message.Reason != monitorapi.DisruptionBeganEventReason {
+			continue // only interested in disruption
+		}
+		if event.To.Before(periodStart) {
+			continue // disruption ended before interval
+		}
+		if event.From.After(periodEnd) {
+			break // no need to examine events entirely later than the period
+		}
+		// we are left with disruption events where some or all is in the period of interest
+		if earlyE2EBackendRe.MatchString(event.Locator.Keys["backend-disruption-name"]) {
+			intervalCount++
+			intervalMessages.Insert(event.Message.HumanMessage)
+		}
+	}
+	if intervalCount > 0 {
+		junits = []*junitapi.JUnitTestCase{
+			{
+				Name: testName,
+				FailureOutput: &junitapi.FailureOutput{
+					Output: fmt.Sprintf("found %d disruption events near E2E test start with messages: %v", intervalCount, intervalMessages.UnsortedList()),
```

Comment:
Lets slip in the time the first e2e test started into the message for the person eventually debugging.